### PR TITLE
AIP-67 - Introduce team id to executor names and loader

### PR DESCRIPTION
**Pull request by o-nikolas**
When multi team (AIP-67) is eventually in place, the scheduler will need to initialize a set of executors for each team. This change is a first step towards that which updates the ExecutorName data model and the executor loader to be "team aware".

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/executors/executor_loader.py
Comment (o-nikolas) [inline airflow/executors/executor_loader.py:181]: Config changes will be coming in a future PR. This will retrieve the fallback of None, and so the method will return an empty list (thus gating this new code). If someone does try to use that config setting (and the name may change) they will hit the exception in `block_use_of_multi_team()`

File: airflow/executors/base_executor.py
Comment (o-nikolas) [inline airflow/executors/base_executor.py:135]: More changes coming in a future PR which will make use of this.

**Comment by o-nikolas** [comment]
CC @potiuk

**Review by shahar1 — COMMENTED** [review]
I assume that it is done as part of [AIP-67](https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-67+Multi-team+deployment+of+Airflow+components) - I think that it's worth mentioning it in the title and contents, as well as creating a [project](https://github.com/apache/airflow/projects?query=is%3Aopen) for that so it will become easier to track.
Before merging, it would be nice to outline a roadmap (general guidelines) so it would be clearer to understand the broader context of this task :)

File: tests/cli/test_cli_parser.py
Comment (ashb) [inline tests/cli/test_cli_parser.py:240]: Is it worth special casing the error message when team id is not present/used?

It's not likely users will hit this, so maybe it doesn't matter

**Review by ashb — APPROVED** [review]

**Review by vincbeck — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Review by eladkal — APPROVED** [review]

File: tests/cli/test_cli_parser.py
Comment (o-nikolas) [reply tests/cli/test_cli_parser.py:240]: Yeah I agree with your second sentiment, it's a pretty unlikely scenario and I think what they get here will be enough. If they did provide a team-name or alias then it will show up in the message as well.

**Comment by o-nikolas** [comment]
> I assume that it is done as part of AIP-67 - I think that it's worth mentioning it in the title and contents, as well as creating a project for that so it will become easier to track.

Yupp this is AIP-67. That's a fair call out, I'll update the description, title and add a tag.
I have a [project plan document here](https://docs.google.com/document/d/11rKo5D2QpT5NvMtDR1RZDjaih5jT5H-dt0aepkfmXSE/edit?usp=sharing) which you can read if you're interested. I planned to also create a project board but am still working through the process of disambiguation and really feeling out how it will be pulled off. Once I have a few of the more foundational topics sorted and I have more clarity I will put a board together.

> Before merging this one, it would be nice to have a roadmap so it would be clearer to understand the broaded context of this task :slightly_smiling_face:

As I said above, I'll work on this, but it has never before been a requirement of merging PRs. And I don't think it's wise to add that barrier to our contribution process. So I don't think it should block this PR's merge.

**Comment by shahar1** [comment]
> > I assume that it is done as part of AIP-67 - I think that it's worth mentioning it in the title and contents, as well as creating a project for that so it will become easier to track.
> 
> Yupp this is AIP-67. That's a fair call out, I'll update the description, title and add a tag. I have a [project plan document here](https://docs.google.com/document/d/11rKo5D2QpT5NvMtDR1RZDjaih5jT5H-dt0aepkfmXSE/edit?usp=sharing) which you can read if you're interested. I planned to also create a project board but am still working through the process of disambiguation and really feeling out how it will be pulled off. Once I have a few of the more foundational topics sorted and I have more clarity I will put a board together.
> 
> > Before merging this one, it would be nice to have a roadmap so it would be clearer to understand the broaded context of this task 🙂
> 
> As I said above, I'll work on this, but it has never before been a requirement of merging PRs. And I don't think it's wise to add that barrier to our contribution process. So I don't think it should block this PR's merge.

I apologize if it was interpreted as a barrier/blocker - it wasn't my intention at all. The doc that you created is exactly what I was looking for, just to understand how it's going to work.

**Pull request merged**
